src/main/scala/org/constellation/snapshotstreaming/opensearch/UpdateRequestBuilder.scala

```diff
@@ -10,89 +10,117 @@ import com.sksamuel.elastic4s.requests.update.UpdateRequest
+        currSnapshot,
+        currIncrementalSnapshots,
+        currBlocks,
+        currTransactions,
+        currFeeTransactions,
+        currBalances)
+    }
+  }
+}
+
 trait UpdateRequestBuilder[F[_]] {
 
   def bulkUpdateRequests(
-    globalSnapshotWithState: GlobalSnapshotWithState,
-    timestamp: Date,
-    hasher: Hasher[F]
-  ): F[UpdateRequests]
+                          splitSnapshot: SplitSnapshot
+                        ): UpdateRequests
 
 }
 
 object UpdateRequestBuilder {
 
   def make[F[_]: Async](
-    globalMapper: GlobalSnapshotMapper[F],
-    currencyMapper: CurrencySnapshotMapper[F],
     config: Configuration,
-    txHasher: Hasher[F]
   ): UpdateRequestBuilder[F] =
     new UpdateRequestBuilder[F] {
 
       def bulkUpdateRequests(
-        globalSnapshotWithState: GlobalSnapshotWithState,
-        timestamp: Date,
-        hasher: Hasher[F]
-      ): F[UpdateRequests] =
-        for {
-          _ <- Async[F].unit
-          GlobalSnapshotWithState(globalSnapshot, maybePrevSnapshotInfo, snapshotInfo, currencySnapshots) =
-            globalSnapshotWithState
-
-          mappedGlobalData <- globalMapper.mapGlobalSnapshot(
-            globalSnapshot,
-            maybePrevSnapshotInfo,
-            snapshotInfo,
-            timestamp,
-            txHasher,
-            hasher
-          )
-          (snapshot, blocks, transactions, balances) = mappedGlobalData
-
-          mappedCurrencyData <- currencyMapper.mapCurrencySnapshots(
-            currencySnapshots,
-            maybePrevSnapshotInfo.map(_.lastCurrencySnapshots),
-            timestamp,
-            txHasher,
-            hasher
-          )
-          (currSnapshot, currIncrementalSnapshots, currBlocks, currTransactions, currFeeTransactions, currBalances) =
-            mappedCurrencyData
-
-          parallelRequests = updateParallelRequests(
-            blocks,
-            transactions,
-            balances,
-            currSnapshot,
-            currIncrementalSnapshots,
-            currBlocks,
-            currTransactions,
-            currFeeTransactions,
-            currBalances
-          ).grouped(config.bulkSize).toList
-
-          sequentialRequests = updateSequentialRequests(
-            snapshot,
-          ).grouped(config.bulkSize).toSeq
-
-        } yield UpdateRequests(
+        splitSnapshot: SplitSnapshot
+      ): UpdateRequests = {
+
+        val SplitSnapshot(snapshot, blocks,
+        transactions,
+        balances,
+        currSnapshot,
+        currIncrementalSnapshots,
+        currBlocks,
+        currTransactions,
+        currFeeTransactions,
+        currBalances) = splitSnapshot
+        val parallelRequests = updateParallelRequests(
+          blocks,
+          transactions,
+          balances,
+          currSnapshot,
+          currIncrementalSnapshots,
+          currBlocks,
+          currTransactions,
+          currFeeTransactions,
+          currBalances
+        ).grouped(config.bulkSize).toList
```

Comment:
maybe you can put whole split snapshot into the `updateParallelRequests` function?